src/saltext/vault/modules/vault_pki.py

```diff
@@ -0,0 +1,1130 @@
+    """
+    endpoint = f"{mount}/revoke/"
+    payload = {}
+
+    if serial is None and certificate is None:
+        raise CommandExecutionError("either serial or certificate must be passed.")
+
+    if serial is not None and certificate is not None:
+        raise CommandExecutionError("only one of serial or certificate must be passed, not both")
+
+    try:
+        if certificate is not None:
+            certificate = x509util.load_cert(certificate)
+            cert_encoding = getattr(serialization.Encoding, "PEM")
+            cert_bytes = certificate.public_bytes(cert_encoding)
+            payload["certificate"] = cert_bytes.decode()
+        elif serial is not None:
+            if isinstance(serial, int):
+                serial = dec2hex(serial)
+            payload["serial_number"] = serial
+
+        vault.query("POST", endpoint, __opts__, __context__, payload=payload)
+        return True
+    except vault.VaultInvocationError:
+        return False
+    except vault.VaultException as err:
+        raise CommandExecutionError(f"{err.__class__}: {err}") from err
+
+
+def read_urls(mount="pki"):
+    """
+    Fetch the URLs to be encoded in generated certificates.
+    No URL configuration will be returned until the configuration is set.
+
+    `API method docs <https://developer.hashicorp.com/vault/api-docs/secret/pki#read-urls>`__.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+            salt '*' vault_pki.get_urls
+
+    mount
+        The mount path the PKI backend is mounted to. Defaults to ``pki``.
+    """
+    endpoint = f"{mount}/config/urls"
+
+    try:
+        return vault.query("GET", endpoint, __opts__, __context__)["data"]
+    except vault.VaultException as err:
+        raise CommandExecutionError(f"{err.__class__}: {err}") from err
+
+
+def _split_sans(sans) -> tuple[list, list, list, list]:
+    dns_sans = []
+    ip_sans = []
+    uri_sans = []
+    other_sans = []
+
+    try:
+        if isinstance(sans, list):
+            sans = dict(map(lambda x: x.split(":", 1), sans))
+
+        for k, v in sans.items():
+            if k.upper() == "DNS" or k.upper() == "EMAIL":
+                dns_sans.append(v)
+            elif k.upper() == "IP":
+                ip_sans.append(v)
+            elif k.upper() == "URI":
+                uri_sans.append(v)
+            else:
+                other_sans.append(f"{k};UTF8:{v}")
+    except ValueError as err:
+        raise CommandExecutionError(
+            f"SAN is not in correct format. Must be in format <type>:<value>: {err}"
+        ) from err
+
+    return dns_sans, ip_sans, uri_sans, other_sans
+
+
+def _build_csr(private_key, private_key_passphrase=None, digest="sha256", **kwargs):
+    if digest.lower() not in [
+        "sha1",
+        "sha224",
+        "sha256",
+        "sha384",
+        "sha512",
+        "sha512_224",
+        "sha512_256",
+        "sha3_224",
+        "sha3_256",
+        "sha3_384",
+        "sha3_512",
```

Comment:
nitpick: This could be a module-global tuple (ironically I think I wrote this code myself).

Reply:
Moved as tuple.